/*
 * This file was automatically generated by EvoSuite
 * Sun Jan 05 22:54:26 GMT 2020
 */

package org.apache.tools.ant.filters;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import java.io.IOException;
import java.io.Reader;
import java.io.StringReader;
import org.apache.tools.ant.filters.PrefixLines;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true, useJEE = true) 
public class PrefixLines_ESTest extends PrefixLines_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      PrefixLines prefixLines0 = new PrefixLines();
      Reader reader0 = prefixLines0.chain(prefixLines0);
      assertNotSame(prefixLines0, reader0);
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      StringReader stringReader0 = new StringReader("*f<U)49~");
      PrefixLines prefixLines0 = new PrefixLines(stringReader0);
      int int0 = prefixLines0.read();
      assertEquals(42, int0);
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      PrefixLines prefixLines0 = new PrefixLines();
      prefixLines0.setPrefix("*f<U)49~");
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      PrefixLines prefixLines0 = new PrefixLines();
      try { 
        prefixLines0.read();
        fail("Expecting exception: IOException");
      
      } catch(IOException e) {
         //
         // Stream closed
         //
         verifyException("java.io.StringReader", e);
      }
  }

  @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      PrefixLines prefixLines0 = new PrefixLines();
      // Undeclared exception!
      try { 
        prefixLines0.chain((Reader) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("java.io.Reader", e);
      }
  }

  @Test(timeout = 4000)
  public void test5()  throws Throwable  {
      PrefixLines prefixLines0 = null;
      try {
        prefixLines0 = new PrefixLines((Reader) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("java.io.Reader", e);
      }
  }
}
